cellrank/kernels/_monge_kernel.py

```diff
@@ -0,0 +1,116 @@
+from typing import Any, Tuple, Union, Literal, Callable, Optional, Sequence
+
+from moscot.problems.time import TemporalNeuralProblem
+
+from cellrank._utils._docs import d, inject_docs
+from scvelo.preprocessing.moments import get_moments
+from cellrank.kernels._velocity_kernel import VelocityKernel
+from cellrank.kernels.utils._similarity import Similarity
+from cellrank.kernels.utils._velocity_model import BackwardMode, VelocityModel
+
+import numpy as np
+
+__all__ = ["MongeKernel"]
+
+_ATTR_OPTIONS = Literal["obsm", "layers"]
+
+
+@d.dedent
+class MongeKernel(VelocityKernel):
+    """
+    Summary.
+
+    Description.
+
+    Args:
+        VelocityKernel (_type_): _description_
+    """
+
+    def __init__(
+        self,
+        tnp: TemporalNeuralProblem,
+        backward: bool = False,
+        attr: _ATTR_OPTIONS = "obsm",
+        xkey: Optional[str] = "X_pca",
+        vkey: Optional[str] = "Monge_velocities",
+        **kwargs: Any,
+    ):
+        super().__init__(
+            tnp.adata,
+            backward=backward,
+            xkey=xkey,
+            vkey=vkey,
+            attr=attr,
+            **kwargs,
+        )
+        self._logits: Optional[np.ndarray] = None
+
+    def _read_from_adata(
+        self,
+        attr: _ATTR_OPTIONS = "obsm",
+        xkey: Optional[str] = "X_pca",
+        vkey: Optional[str] = "Monge_velocities",
+        gene_subset: Optional[Union[str, Sequence[str]]] = None,
+        **kwargs: Any,
+    ) -> None:
+        super(VelocityKernel, self)._read_from_adata(**kwargs)
+
+        if attr == "layers":
+            self._xdata = self._extract_layer(xkey, subset=gene_subset)
```

Comment:
I think it might be better to generalize `_extract_layers` to also handle `attr` information rather than overriding `_read_from_adata` (lot's of code repeat itself here). Then in `VelocityKernel`, `attr` will not be exposed in `__init__`, only in `_read_from_adata` it will have the default `layers`.